pkg/controller/controller_broker.go

```diff
@@ -540,11 +541,27 @@ func (c *controller) reconcileClusterServiceClassFromClusterServiceBrokerCatalog
 	toUpdate.Spec.ExternalName = serviceClass.Spec.ExternalName
 	toUpdate.Spec.ExternalMetadata = serviceClass.Spec.ExternalMetadata
 
-	if _, err := c.serviceCatalogClient.ClusterServiceClasses().Update(toUpdate); err != nil {
+	updatedServiceClass, err := c.serviceCatalogClient.ClusterServiceClasses().Update(toUpdate)
+	if err != nil {
 		glog.Error(pcb.Messagef("Error updating %s: %v", pretty.ClusterServiceClassName(serviceClass), err))
 		return err
 	}
 
+	if updatedServiceClass.Status.RemovedFromBrokerCatalog {
+		glog.V(4).Info(pcb.Messagef("Resetting RemovedFromBrokerCatalog status on %s", pretty.ClusterServiceClassName(serviceClass)))
+		updatedServiceClass.Status.RemovedFromBrokerCatalog = false
```

Comment:
I wonder if we can check somehow that this is the same class and not a completely new one with a clashing name.

Reply:
OSB spec is not clear about whether the name can be reused after the service was deleted from the catalog:
https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#service-object

| Response Field | Type | Description |
| --- | --- | --- |
| name* | string | A CLI-friendly name of the service. MUST only contain alphanumeric characters and hyphens (no spaces). MUST be unique across all service objects returned in this response. MUST be a non-empty string. |

It only says that the name must be unique within a response, which implicitly allows to redeclare a new service with a clashing name once the old one was removed.

Reply:
Ah we use `ID` as a Kubernetes resource name there, which must be globally unique. All good 👍